include/nanobind/nb_types.h

```diff
@@ -443,6 +443,33 @@ class bytes : public object {
     size_t size() const { return (size_t) PyBytes_Size(m_ptr); }
 };
 
+class bytearray : public object {
+    NB_OBJECT(bytearray, object, "bytearray", PyByteArray_Check)
+
+    bytearray()
+        : object(PyObject_CallNoArgs((PyObject *)&PyByteArray_Type), detail::steal_t{}) { }
```

Comment:
It appears from CI that `PyObject_CallNoArgs` doesn't exist in 3.8. You might need an #ifdef here. The alternative would be `PyObject_CallObject` with a second argument of NULL. `CallNoArgs` is documented as faster on the versions where it exists, so we should probably keep it for 3.9 and later.

Reply:
I'll mark this as done when CI completes, though I also tested it manually.